components/gallery/GalleryItemAction/GalleryItemActionType/GalleryItemBuy.vue

```diff
@@ -1,50 +1,178 @@
+              :disabled="disabled"
+              no-shadow
+              @click.native="onClick" />
+          </NeoTooltip>
+        </template>
+
+        <template #content>
+          <div class="has-text-centered">
+            {{ $t('nft.buyNFTOn') }}
+            <span class="has-text-weight-bold is-uppercase">{{
+              urlPrefix
+            }}</span>
+          </div>
+        </template>
+      </GalleryItemActionSlides>
+      <div v-else>{{ $t('nft.notListed') }}</div>
+    </GalleryItemPriceSection>
+  </div>
 </template>
 
 <script setup lang="ts">
-import { NeoButton } from '@kodadot1/brick'
-import { onClickOutside } from '@vueuse/core'
-
-import { useGalleryItem } from '../../useGalleryItem'
+import { NeoButton, NeoTooltip } from '@kodadot1/brick'
 import GalleryItemPriceSection from '../GalleryItemActionSection.vue'
 import GalleryItemActionSlides from '../GalleryItemActionSlides.vue'
+import { onClickOutside } from '@vueuse/core'
+import { notificationTypes, showNotification } from '@/utils/notification'
+import { getApiCall } from '@/utils/gallery/abstractCalls'
+import { getKusamaAssetId } from '@/utils/api/bsx/query'
+import { somePercentFromTX } from '@/utils/support'
+import { tokenIdToRoute } from '@/components/unique/utils'
+import { JustInteraction, createInteraction } from '@kodadot1/minimark'
+import nftByIdMinimal from '@/queries/rmrk/subsquid/nftByIdMinimal.graphql'
+const props = defineProps<{
+  nftId: string
+  currentOwner: string
+  collectionId: string
+  nftPrice: string
+}>()
+
+const { urlPrefix, client } = usePrefix()
+const { accountId } = useAuth()
+const { $store, $apollo, $i18n } = useNuxtApp()
+const { apiInstance } = useApi()
+const ACTION = 'BUY'
+const actionLabel = $i18n.t('nft.action.buy')
 
-const { urlPrefix } = usePrefix()
-const { nft } = useGalleryItem()
+const { howAboutToExecute, initTransactionLoader, isLoading, status } =
+  useMetaTransaction()
 
 const active = ref(false)
-const label = computed(() => (active.value ? 'Confirm' : 'Buy'))
+const label = computed(() =>
+  active.value ? $i18n.t('nft.action.confirm') : $i18n.t('nft.action.buy')
+)
+
+const balance = computed<string>(() => {
+  if (urlPrefix.value == 'rmrk') {
+    return $store.getters.getAuthBalance
+  }
+  return $store.getters.getTokenBalanceOf(getKusamaAssetId(urlPrefix.value))
+})
+const disabled = computed(() => {
+  if (!(props.nftPrice && balance.value)) {
+    return false
+  }
+  return Number(balance.value) <= Number(props.nftPrice)
+})
+
+function onClick() {
+  if (active.value) {
+    handleBuy()
+  } else {
+    active.value = true
+  }
+}
+// close the buy button when transaction loading is finsihed
+watch(isLoading, (loading) => {
+  active.value = loading
+})
+
+const getTranasactionParams = async () => {
+  const api = await apiInstance.value
+  if (urlPrefix.value == 'rmrk') {
+    const rmrk = createInteraction(
+      ACTION as JustInteraction,
+      useRmrkVersion().version,
```

Comment:
`useRmrkVersion` who's this?

Reply:
nvm I didn't saw composable, why do you need this for? `RmrkVersionMixin` can't be used?

Reply:
mmm because this component is not a class component?

Reply:
Aight Vue3 no mixin!